pageserver/src/page_service.rs

```diff
@@ -350,9 +369,60 @@ impl PageServerHandler {
 
         Ok(())
     }
+
+    // when accessing management api supply None as an argument
+    // when using to authorize tenant pass corresponding tenant id
+    fn check_permission(&self, tenantid: Option<ZTenantId>) -> Result<()> {
+        if self.auth.is_none() {
+            // auth is set to Trust, nothing to check so just return ok
+            return Ok(());
+        }
+        let claims = self.claims.as_ref().unwrap();
```

Comment:
I'll try to figure out how to remove this unwrap by proving claims existence in this case using types. Currently this unwrap shouldn't trigger because during connection initialisation if auth is set to ZenithJWT claims will always be Some

Reply:
Thinking a bit about type safe version, and all my thoughts incur introduction of a dedicated auth trait and a bunch of other types for states and transitions for use in postgres backend, and I'm not sure how ergonomic it would be, since it will need to support both pageserver and proxy requirements (potentially with later scram support in proxy). And to me It looks like it is not worth it now. So I'm not sure about implementing this kind of thing in here, but I'll try to make some sketches about how this could look like

So for now I've just added a comment describing why this unwrap/expect shouldn't be triggered